src/webapp/src/auth/Auth.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import styled from 'styled-components';
+import { useDispatch, useSelector } from 'react-redux';
+import { AuthStatuses } from './types';
+import { RootState } from '../reducers';
+import { signInUser, signOutUser } from './asyncAuthActions';
+
+const Loader = styled.div`
+  border: 12px solid #f3f3f3;
+  border-radius: 50%;
+  border-top: 12px solid #48b0f5;
+  width: 30px;
+  height: 30px;
+  -webkit-animation: spin 2s linear infinite;
+  animation: spin 2s linear infinite;
+  @keyframes spin {
+    0% {
+      transform: rotate(0deg);
+    }
+    100% {
+      transform: rotate(360deg);
+    }
+  }
+`;
+
+export const Auth = () => {
+  const dispatch = useDispatch();
+  const { authStatus, user, error } = useSelector(
+    (state: RootState) => state.user
+  );
+
+  const handleLogout = () => {
+    dispatch(signOutUser());
+  };
+
+  if (error) {
+    return (
+      <p>There has been an error: ${error}. Refresh the page and try again</p>
+    );
+  }
+
+  if (authStatus === AuthStatuses.loading) {
+    return <Loader />;
+  }
+
+  return (
+    <div>
+      {authStatus === AuthStatuses.loggedIn ? (
+        <p>
+          Welcome: {user?.displayName} your id is {user?.uid} and you are using
```

Comment:
I was checking in case a user wouldn't have one of those properties. I suppose in case of the `uid` and the `email` it doesn't make sense because they are going to be there always, just `displayName` is optional when we have other sing up methods, which we don't have now. So I suppose it is all completely unnecessary. Thanks!